doc/developers/advanced_installation.rst

```diff
@@ -22,127 +33,191 @@ basis to help users test bleeding edge features or bug fixes::
 Building from source
 =====================
 
-In the vast majority of cases, building scikit-learn for development purposes
-can be done with::
+.. _git_repo:
 
-    pip install cython pytest flake8
+Use `Git <https://git-scm.com/>`_ to check out the latest source from the
+`scikit-learn repository <https://github.com/scikit-learn/scikit-learn>`_ on
+`Github <https://github.com>`_ and pip to build the project::
 
-Then, in the main repository::
+    git clone git://github.com/scikit-learn/scikit-learn.git
+    cd scikit-learn
 
+    pip install cython
     pip install --editable .
 
-Please read below for details and more advanced instructions.
+Check that the installed scikit-learn with a version number ending with
+`.dev0`::
+
+    python -c "import sklearn; sklearn.show_versions()"
+
+Run the tests with ``pytest``::
+
+    pip install pytest
+    pytest sklearn
```

Comment:
I would not add this, we don't want contributors to think `pytest sklearn` is the proper way to test their changes.

IMO the small snippet above is enough to ensure the install went correctly

Reply:
What would you recommend to launch the tests instead? Just "pytest" without `sklearn`?

Reply:
I would just not mention tests here.

If you think we should, maybe link to point 2 of https://scikit-learn.org/dev/developers/contributing.html#pull-request-checklist

Reply:
Also I'll make a PR to update the testing section of the contributing guide because we recommend using `make` which is wrong

Reply:
see https://github.com/scikit-learn/scikit-learn/pull/15377

Reply:
I have updated the PR to link to the developers guide and the pytest tips.